### PR TITLE
[K8S] Store secret - fix bug when data is None

### PR DESCRIPTION
**Pull request by roei3000b**
When we store an empty secret, the `k8s_secret.data` will be `None`.
The bug was that we tried to call to `copy` function on `k8s_secret.data` but it was None so we got `AttributeError`.

https://iguazio.atlassian.net/browse/ML-7356

### CODE REVIEW

File: server/api/utils/singletons/k8s.py [hunk 1]
```diff
@@ -533,7 +533,7 @@ def store_secrets(
                     )
                 raise exc
 
-        secret_data = k8s_secret.data.copy()
+        secret_data = (k8s_secret.data or {}).copy()
```
Comment (TomerShor) [inline server/api/utils/singletons/k8s.py]: you don't need to copy an empty dict, so how about this?
```suggestion
        secret_data = k8s_secret.data.copy() of k8s_secret.data else {}
```

Comment (roei3000b) [reply server/api/utils/singletons/k8s.py]: You right about no need to copy an empty dict but I think the way I did it is more compact.

Comment (TomerShor) [reply server/api/utils/singletons/k8s.py]: yes it's compact, but you created 2 objects instead of one.

**Review by alonmr — COMMENTED** [review]
can you explain why the secret was created? I believe we shouldn't have empty project secrets or is there a flow I'm not thinking of?
Also if you get a chance please fix the docstring
```
:param secrets:     the secrets to delete
```

**Comment by roei3000b** [comment]
> can you explain why the secret was created? I believe we shouldn't have empty project secrets or is there a flow I'm not thinking of? Also if you get a chance please fix the docstring
> 
> ```
> :param secrets:     the secrets to delete
> ```

`db.create_project_secrets(project="khsfcrscud", provider=SecretProviderName.kubernetes, secrets={})`
(from the [notebook](https://iguazio.atlassian.net/browse/ML-7356))

**Review by alonmr — COMMENTED** [review]
what about https://github.com/mlrun/mlrun/blob/development/server/api/utils/singletons/k8s.py#L600

**Comment by moranbental** [comment]
> what about https://github.com/mlrun/mlrun/blob/development/server/api/utils/singletons/k8s.py#L600

This is another bug that I am working on - https://github.com/mlrun/mlrun/pull/6111

File: server/api/utils/singletons/k8s.py [hunk 2]
```diff
@@ -533,7 +533,7 @@ def store_secrets(
                     )
                 raise exc
 
-        secret_data = k8s_secret.data.copy()
+        secret_data = k8s_secret.data.copy() if k8s_secret.data else {}
         for key, value in secrets.items():
```
Comment (moranbental) [inline server/api/utils/singletons/k8s.py:542]: If the `secrets` parameter is `None`, the code will raise an error when attempting to iterate over `secrets.items()`

**Review by moranbental — CHANGES_REQUESTED** [review]

File: server/api/utils/singletons/k8s.py [hunk 3]
```diff
@@ -533,9 +533,10 @@ def store_secrets(
                     )
                 raise exc
 
-        secret_data = k8s_secret.data.copy()
-        for key, value in secrets.items():
-            secret_data[key] = base64.b64encode(value.encode()).decode("utf-8")
+        secret_data = k8s_secret.data.copy() if k8s_secret.data else {}
+        if secrets:
```
Comment (moranbental) [inline server/api/utils/singletons/k8s.py]: Now, if `secrets` is `None`, we are still sending an update event, even though no update should be made. The function should first check if the `secrets` parameter is `None` and return `None` if it is. Only if `secrets` is not `None` should the function proceed with `secret_data = k8s_secret.data.copy() if k8s_secret.data else {}` and call `replace_namespaced_secret`.

I also suggest adding some unit tests for this

**Review by moranbental — CHANGES_REQUESTED** [review]

**Review by moranbental — APPROVED** [review]
Great!

**Review by liranbg — APPROVED** [review]

**Pull request merged**